### PR TITLE
Fix color valid color mix test

### PR DESCRIPTION
**Pull request by mysteryDate**
color-mix values should not get resolved at specified value time. See: https://github.com/web-platform-tests/wpt/issues/34158 and https://github.com/w3c/csswg-drafts/issues/7302#issuecomment-1153984819

This PR changes keeps all the cases in the current test, but does not expect the color mix to resolve, only to resolve the input colors as they normally would at specified value time (i.e. `#ff00ff` -> `rgb(255, 0, 255)`). 

There are definitely some cases that are missing here: `calc` values, keyword colors, current color, light dark pairs and hex colors. I'll add those next.

### CODE REVIEW

**Comment by mysteryDate** [comment]
@lilles

**Review by svgeesus — COMMENTED** [review]
This looks good to me.

I noticed in reviewing that the test assumes lab and oklab, lch and oklch have similar ranges which they don't. But that could be fixed in a separate PR.

File: css/css-color/parsing/color-valid-color-mix-function.html [hunk 1]
```diff
@@ -15,326 +15,310 @@
...
-    test_valid_value(`color`, `color-mix(in hwb, lch(0 116 334) 100%, rgb(0, 0, 0) 0%)`, `rgb(42, 0, 34)`); // Naive clip based mapping would give rgb(90, 0, 76). NOTE: 0% lightness in Lab/LCH does not automatically correspond with sRGB black,
-    test_valid_value(`color`, `color-mix(in hwb, oklab(100 0.365 -0.16) 100%, rgb(0, 0, 0) 0%)`, `rgb(255, 255, 255)`); // Naive clip based mapping would give rgb(255, 92, 255).
-    test_valid_value(`color`, `color-mix(in hwb, oklab(0 0.365 -0.16) 100%, rgb(0, 0, 0) 0%)`, `rgb(0, 0, 0)`); // Naive clip based mapping would give rgb(19, 0, 24).
-    test_valid_value(`color`, `color-mix(in hwb, oklch(100 0.399 336.3) 100%, rgb(0, 0, 0) 0%)`, `rgb(255, 255, 255)`); // Naive clip based mapping would give rgb(255, 91, 255).
-    test_valid_value(`color`, `color-mix(in hwb, oklch(0 0.399 336.3) 100%, rgb(0, 0, 0) 0%)`, `rgb(0, 0, 0)`); // Naive clip based mapping would give rgb(20, 0, 24).
+    test_valid_value(`color`, `color-mix(in hsl, hsl(120deg 10% 20%), hsl(30deg 30% 40%))`, `color-mix(hsl, rgb(46, 56, 46), rgb(133, 102, 71))`);
```
Comment (lilles) [inline css/css-color/parsing/color-valid-color-mix-function.html:18]: Why is the 'is' keyword left out of the serialization? It seems to be required by the spec?

**Review by lilles — APPROVED** [review]
There are a lot of changes here, but generally lgtm. Firefox seems to have a pretty good impl in this area so if you could check if Firefox passes all, or if any failures are expected, before landing, that would make me more confident about these changes.

**Comment by mysteryDate** [comment]
I actually based these changes on Firefox. It still fails most of them because they do not yet handle "none" and is doing some computed-value stuff to the percentages that's not in the spec. Otherwise it matches.

**Comment by dbaron** [comment]
Looks like this was basically ready to merge, but it now has conflicts.  Is it worth fixing and merging (and maybe addressing @svgeesus 's comment while doing so)?

**Comment by svgeesus** [comment]
Yes, it is worth re-basing to fix the conflicts.

**Comment by mysteryDate** [comment]
Unfortunately git is not handling diffs very well. The is a nasty merge, I'll see what I can do.

**Comment by mysteryDate** [comment]
They are resolved!

**Comment by mysteryDate** [comment]
We're having an issue where for a bunch of tests chrome and safari return `127` for the blue channel, though the upstream changes to this file expect `128`.

For `hwb(320deg 30% 40%)` using the algorithm `hwbToRgb` as defined in this document, the result is:
`[0.5999999999999999, 0.3, 0.5]`

So the issue is that chrome and safari are evaluating `0.5` to an integer in the range `[0,255]` as `127`, whereas the test expects `128`. Which one is correct? Should the test accept either. This test is already doing a lot, we should probably test integer rounding somewhere else.

**Comment by svgeesus** [comment]
> I think in the future we should try have smaller test files to avoid issues like this

That was how things _were_ (small, focused tests for specific, small specification sections) until there was a [massive rewrite](https://github.com/web-platform-tests/wpt/pull/36353) which consolidated things into one-test-per-property style.

> So the issue is that chrome and safari are evaluating 0.5 to an integer in the range [0,255] as 127, whereas the test expects 128. Which one is correct? Should the test accept either. This test is already doing a lot, we should probably test integer rounding somewhere else.

From CSS Color 4 5.1. The RGB functions: [rgb()](https://drafts.csswg.org/css-color-4/#funcdef-rgb) and [rgba()](https://drafts.csswg.org/css-color-4/#funcdef-rgba) which in turn references CSS Values 4 on rounding:

> These values come from the fact that many graphics engines store the color channels internally as a single byte, which can hold integers between 0 and 255. Implementations should honor the precision of the channel as authored or calculated wherever possible. If this is not possible, the channel should be [rounded towards +∞](https://drafts.csswg.org/css-values-4/#combine-integers).

```js
a = 0.5 * 255
127.5
b = Math.round(a)
128 
```
From [round to the nearest integer](https://drafts.csswg.org/css-values-4/#css-round-to-the-nearest-integer):

> Unless otherwise specified, in the CSS specifications rounding to the nearest integer requires rounding in the direction of +∞ when the fractional portion is exactly 0.5. (For example, 1.5 rounds to 2, while -1.5 rounds to -1.)

**Comment by svgeesus** [comment]
> we should probably test integer rounding somewhere else.

It should probably be tested more generally in css/css-values, but here it is testing a specific reference from CSS Color 4.

**Comment by dbaron** [comment]
Closing and reopening to re-trigger running of checks, since it looks like the underlying error was:
```
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://hg.mozilla.org/mozilla-central/archive/tip.zip/testing/profiles/
```

**Comment by mysteryDate** [comment]
Okay, I've set this to round to 128 and not 127, I'll investigate why chrome is doing the wrong thing here.

**Comment by mysteryDate** [comment]
> It should probably be tested more generally in css/css-values, but here it is testing a specific reference from CSS Color 4.

Isn't it just testing how "0.5 * 255" is rounded when converting `hwb(320deg 30% 40%)` to rgb?
https://jsfiddle.net/6yfqnLhs/

**Comment by mysteryDate** [comment]
> Isn't it just testing how "0.5 * 255" is rounded when converting hwb(320deg 30% 40%) to rgb?
> https://jsfiddle.net/6yfqnLhs/

I'm adding a simple parsing test to check this here:
https://chromium-review.googlesource.com/c/chromium/src/+/4507865

> That was how things were (small, focused tests for specific, small specification sections) until there was a https://github.com/web-platform-tests/wpt/pull/36353 which consolidated things into one-test-per-property style.

It would be my instinct to call this test something like `hwb-float-to-integer-rounding.html` but to reflect the structure as is, I just added it to `color-valid-hwb.html`, at least with a comment explaining the intent of the test.

**Pull request merged**